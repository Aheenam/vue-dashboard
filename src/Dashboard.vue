<template>
  <div class="dashboard" :style="grid">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'dashboard',
  props: {
    rows: {
      type: Number,
      default: 3
    },
    columns: {
      type: Number,
      default: 5
    }
  },
  data () {
    return {
      grid: {
        gridTemplateColumns: '1fr '.repeat(this.columns),
        gridTemplateRows: '1fr '.repeat(this.rows)
      }
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>

.dashboard {
  display: grid;
  grid-column-gap: 1rem;
  grid-row-gap: 1rem;
  height: 100%;
}

</style>
