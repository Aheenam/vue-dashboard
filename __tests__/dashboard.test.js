import Dashboard from '../src/Dashboard'
import Vue from 'vue'

describe('dashboard-test', () => {

    const Constructor = Vue.extend(Dashboard);
    const vm = new Constructor().$mount()

    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    });

})
