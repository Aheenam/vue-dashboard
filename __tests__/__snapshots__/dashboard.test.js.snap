// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard-test should match the snapshot 1`] = `
<div
  class="dashboard"
/>
`;
